src/lib/annotations/Annotator.js

```diff
@@ -541,25 +559,67 @@ class Annotator extends EventEmitter {
+     * Binds event listeners for annotation modes.
      *
      * @protected
+     * @param {string} mode - Current annotation mode
      * @return {void}
      */
-    bindPointModeListeners() {
-        const pointFunc = this.pointClickHandler.bind(this.annotatedElement);
-        const handlers = [
-            {
-                type: 'mousedown',
-                func: pointFunc,
-                eventObj: this.annotatedElement
-            },
-            {
-                type: 'touchstart',
-                func: pointFunc,
-                eventObj: this.annotatedElement
+    bindModeListeners(mode) {
+        let handlers = [];
+
+        if (mode === TYPES.point) {
+            const pointFunc = this.pointClickHandler.bind(this.annotatedElement);
+            handlers = [
+                {
+                    type: 'mousedown',
+                    func: pointFunc,
+                    eventObj: this.annotatedElement
+                },
+                {
+                    type: 'touchstart',
+                    func: pointFunc,
+                    eventObj: this.annotatedElement
+                }
+            ];
+        } else if (mode === TYPES.draw) {
+            const drawingThread = this.createAnnotationThread([], {}, TYPES.draw);
+            const startCallback = drawingThread.handleStart.bind(drawingThread);
+            const stopCallback = drawingThread.handleStop.bind(drawingThread);
+            const moveCallback = drawingThread.handleMove.bind(drawingThread);
+            /* eslint-disable require-jsdoc */
+            const locationFunction = (event) => this.getLocationFromEvent(event, TYPES.point);
+            /* eslint-enable require-jsdoc */
+            const postButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_POST);
+            handlers = [
+                {
+                    type: 'mousemove',
+                    func: annotatorUtil.eventToLocationHandler(locationFunction, moveCallback),
+                    eventObj: this.annotatedElement
+                },
+                {
+                    type: 'mousedown',
+                    func: annotatorUtil.eventToLocationHandler(locationFunction, startCallback),
+                    eventObj: this.annotatedElement
+                },
+                {
+                    type: 'mouseup',
+                    func: annotatorUtil.eventToLocationHandler(locationFunction, stopCallback),
+                    eventObj: this.annotatedElement
+                }
+            ];
+
+            if (postButtonEl) {
+                handlers.push({
+                    type: 'click',
+                    func: () => {
+                        drawingThread.saveAnnotation(TYPES.draw);
+                        this.toggleAnnotationHandler(TYPES.draw);
+                    },
+                    eventObj: postButtonEl
+                });
             }
-        ];
+        }
 
         handlers.forEach((handler) => {
```

Comment:
for each handler in handlers, we need to push the object to this.annotationModeHandler. That way it will be unbound when unbindModeHandlers is called. Handlers is a local array.